### DISCUSSION TITLE
[HPM] Error occurred while trying to proxy request

### DISCUSSION THREAD

**Discussion by jcpeden** [chimurai/http-proxy-middleware#809] [Q&A]
A project I've been working on for a client was working perfectly until about 1 week ago when it randomly stopped working. Now, whenever I try to run the application I get the following error:

`[HPM] Error occurred while trying to proxy request /<sub_dir>/<sub_dir>/<sub_dir>/<sub_dir>/sdk.js from 0.0.0.0:8080 to https://<sub_domain>.<target>com (ECONNRESET) (https://nodejs.org/api/errors.html#errors_common_system_errors)`

The application starts up but as the Node process exists as a result of this, if I refresh the page it dies. The application works perfectly for 3x other team members who have the same corporate proxy permission, Node, NPM and OSX versions as I have. I'm also running a clean pull of our `master` branch with no local changes.

Other SPAs that had been working (and continue to work for other developers) also fail due to HPM errors but the package is unchanged and is still working for other team members.

Strangely, I can access sdk.js (the failing asset above) in a web browser without issue or curl it from terminal. It's usually this asset that causes the failure but sometimes I'll see other proxied requests failing instead.

My tech lead and I tried to debug using onRequest/OnResponse/onError etc and while the onError method seemed to allow the app to fail gracefully, it didn't lead us to anything noteworthy in terms of pinning down where the problem lay.

**Any suggestions on how to debug this? My next step is to ask our service desk for a brand new machine to try!**

### Update
Re-ran the app on a version of Node/NPM that I know works for my colleague and got another HPM error but a slightly different one:

`[HPM] Error occurred while trying to proxy request  from 0.0.0.0:8080 to https://<sub_domain>.<domain>/components/ (ECONNRESET) (https://nodejs.org/api/errors.html#errors_common_system_errors)`

Behaviour is the same in that the app starts but crashes if I refresh the page. Re-ran the application a second time and hit the error mentioned in my first post.

Very odd.

**Comment by chimurai** [comment]
You've got a nice excuse to get a brand new machine 😁

Something changed in your environment... The question is what changed...

You could try WireShark to do some low-level analysis: 
https://medium.com/@shriramsharma/analyze-tcp-connection-resets-with-wireshark-30af498d04fa
